app/src/main/java/com/geekskool/dhobi/Providers/DataProvider.java

```diff
@@ -0,0 +1,136 @@
+package com.geekskool.dhobi.Providers;
+
+import android.content.ContentProvider;
+import android.content.ContentValues;
+import android.content.UriMatcher;
+import android.database.Cursor;
+import android.database.sqlite.SQLiteDatabase;
+import android.net.Uri;
+import android.support.annotation.Nullable;
+
+import com.geekskool.dhobi.Contracts.Contract;
+import com.geekskool.dhobi.DbHelper;
+
+/**
+ * Created by manisharana on 17/2/17.
+ */
+public class DataProvider extends ContentProvider {
```

Comment:
This class makes sense for now, but you would quite quickly want to reconsider your data access layer. Something more robust and modular would help avoid the pitfalls of growing a data access layer organically -- this activity has been played out on every software project ever, so a little bit of "premature optimization" / "design up front" is okay here.